src/js/pages/TwitterHandleLanding.jsx

```diff
@@ -52,35 +52,17 @@ export default class TwitterHandleLanding extends Component {
   }
 
   // eslint-disable-next-line camelcase,react/sort-comp
-  UNSAFE_componentWillReceiveProps (nextProps) {
-    // console.log('TwitterHandleLanding componentWillReceiveProps');
-    let activeRoute = '';
-    let nextTwitterHandle;
-    // eslint-disable-next-line react/prop-types
-    if (nextProps.match && nextProps.location) {
-      // eslint-disable-next-line react/prop-types,camelcase
-      const { location: { pathname }, match: {params: { twitter_handle} }  } = nextProps;
-      activeRoute = pathname;
-      nextTwitterHandle = twitter_handle;
-      // eslint-disable-next-line react/prop-types
-    } else if (this.props && this.props.location) {
-      // eslint-disable-next-line react/prop-types
-      const { location: { pathname } } = this.props;
-      activeRoute = pathname;
-    }
-
-    const { twitterHandle } = this.state;
-    this.setState({
-      activeRoute,
-    });
-    // console.log('TwitterHandleLanding componentWillReceiveProps activeRoute:', activeRoute);
-    if (nextTwitterHandle && twitterHandle.toLowerCase() !== nextTwitterHandle.toLowerCase()) {
-      // We need this test to prevent an infinite loop
-      console.log('TwitterHandleLanding componentWillReceiveProps, different twitterHandle: ', nextProps.match.params.twitter_handle);
+  componentDidUpdate(prevProps) {
```

Comment:
It seems like it would be more efficient to only call:
      TwitterActions.resetTwitterHandleLanding();
      TwitterActions.twitterIdentityRetrieve(nextTwitterHandle);
...when nextTwitterHandle changes, as opposed to also making that call when the pathname changes. I recommend splitting those two different conditions into two blocks. (and two different setStates as well)

Reply:
Alright, I'll make those changes